lib/pulsar/consumer_configuration.rb

```diff
@@ -26,22 +26,22 @@
 module Pulsar
   class ConsumerConfiguration
     # aligns with the pulsar::ConsumerType enum in the C++ library
-    CONSUMER_TYPES = [
-      :exclusive,
-      :shared,
-      :failover,
-      :key_shared
-    ]
+    CONSUMER_TYPES = {
+      :exclusive => Pulsar::ConsumerType::Exclusive,
+      :shared => Pulsar::ConsumerType::Shared,
+      :failover => Pulsar::ConsumerType::Failover,
+      :key_shared => Pulsar::ConsumerType::KeyShared
+    }
 
     module RubySideTweaks
       def consumer_type
         enum_value = super
-        CONSUMER_TYPES[enum_value]
+        CONSUMER_TYPES.invert[enum_value]
       end
 
       def consumer_type=(type)
         unless type.is_a?(Integer)
```

Comment:
Hmm.... This should probably change to `if type.is_a?(Symbol)` to allow passing `Pulsar::ConsumerType::*` explicitly, rather than Integer

Reply:
oh, I missed that in the review, but yes, that needs to change

Reply:
not just to allow passing the enum value directly, but avoid getting the opaque C++ error if an integer is passed (should get the ruby ArgumentError instead)

Reply:
I'd say for the sake of forcing the ArgumentError for e.g. ints and such, I'd keep as an `unless` and make the type checked for be `Pulsar::ConsumerType`. that's working for me in my local version